src/gluonts/dataset/common.py

```diff
@@ -259,17 +259,16 @@ def load_files(
         freq: str,
         chunk_size: int = 100,
     ):
-        paths = util.resolve_paths(paths) if paths is not None else None
-        files = util.find_files(paths, lambda p: str(p).endswith(".arrow"))
-        assert (
-            len(files) > 0
-        ), f"Could not find any arrow files in paths: {paths}"
+        paths = util.resolve_paths(paths)
+        files = util.find_files(paths, lambda p: p.suffix == ".arrow")
+        assert files, f"Could not find any arrow files in paths: {paths}"
         return MemmapArrowDataset(files, freq, chunk_size)
 
     def __iter__(self) -> Iterator[DataEntry]:
-        row_number = 0
         bounds = get_bounds_for_mp_data_loading(len(self))
-        for rec in self.reader.iter_slice(bounds.lower, bounds.upper):
+        for row_number, rec in enumerate(
+            self.reader.iter_slice(bounds.lower, bounds.upper)
```

Comment:
Should maybe be `enumerate(..., start=bounds.lower)` ?

Reply:
@vafl did something change in the logic? Previously you started the `now_number` at 0. Why do you want to start it now at `bounds.lower`? I mean, if you say that's the correct way to implement it, I'll do so, I just want to point out that it seems to be inconsistent with your previous solution.

Reply:
Yes, I think my previous version was wrong. Its not super important, but the purpose of the row number is to track where the data came from and I think it makes more sense if its the row number in the full table.

Reply:
Ok, pushed it, thanks for the explanation!